### PR TITLE
Curse home2

### PR DESCRIPTION
**Pull request by RaduMiu**
missing the right maps, the right pdf rule and design photos

### CODE REVIEW

File: client/src/components/CurseHome.tsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+                >
+                  <span className="font-bold lg:text-24 md:text-20 sm:text-20 text-white  whitespace-nowrap">
+                    MASCULIN
+                  </span>
+                  <br />
+                  <span className="lg:text-16 md:text-14 text-white whitespace-nowrap">
+                    17-35 ANI
+                  </span>
+                </button>
+                <button
```
Comment (MDGcode) [inline client/src/components/CurseHome.tsx:186]: Too much padding on buttons, also add more margin between buttons

File: client/src/components/CurseHome.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+          "https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2848.878166115072!2d26.051764615851956!3d44.43970637910117!2m3!1f0!2f0!3f0!3m2!1d44.43998527916441!2d26.051764615851956!4f0!5f0.7820865974627469"
+        );
+        break;
+      default:
+        break;
+    }
+  };
+
+  return (
+    <div className="w-full px-[var(--bs-gutter-x,.75rem)]   bg-customBackground  ">
```
Comment (MDGcode) [inline client/src/components/CurseHome.tsx:104]: The component is out of proportions
https://imgur.com/a/f2X8RpQ

File: client/src/components/CurseHome.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        setMinimumDonation("Donatie minima: 40 RON");
+        setMapSrc(
+          "https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2848.878166115072!2d26.051764615851956!3d44.43970637910117!2m3!1f0!2f0!3f0!3m2!1d44.43998527916441!2d26.051764615851956!4f0!5f0.7820865974627469"
+        );
+        break;
+      case "FEMININ1":
+        setStartTime("Ora de start: 12:00");
+        setDistance("DISTANȚĂ: 4KM (2 x 2KM)");
+        setMinimumDonation("Donatie minima: 40RON");
+        setMapSrc(
```
Comment (MDGcode) [inline client/src/components/CurseHome.tsx:70]: Same problem with 
https://imgur.com/5P2NtBV

File: client/src/components/CurseHome.tsx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+  return (
+    <div className="w-full px-[var(--bs-gutter-x,.75rem)]   bg-customBackground  ">
+      {/* <img src={flyingImg4} className='float-right top-0 right-0 h-full md:w-1/2 lg:w-1/3 xl:w-1/4 z-0' alt='Imagine de fundal' ></img>  */}
+      <div className="flex justify-center ">
+        <div className="container  md:pt-44  lg:px-32 mx-auto  justify-center">
+          <h1 className="lg:text-52 text-36 font-bold text-center mb-8 text-black">
+            CURSE
+          </h1>
+          <div className="md:flex md:items-center  justify-center">
+            <div className="md:w-1/4 mb-4 md:mb-0 md:mr-4 overflow-x-auto">
```
Comment (MDGcode) [inline client/src/components/CurseHome.tsx:112]: Make it so only 3 buttons are visible at a time on mobile

**Review by MDGcode — CHANGES_REQUESTED** [review]

File: client/src/components/CurseHome.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+  const [minimumDonation, setMinimumDonation] = useState('Donatie minima: 20 RON');
+  const [mapSrc, setMapSrc] = useState("https://www.google.com/maps/embed?pb=!1m42!1m12!1m3!1d1051.7564164316468!2d26.050924936618955!3d44.43943916368305!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!4m27!3e2!4m5!1s0x40b201c187a44aeb%3A0x776c5045f461c6b0!2sBucure%C5%9Fti%20Sectorul%206%2C%20Bucure%C8%99ti%20061344!3m2!1d44.4396542!2d26.052221199999998!4m3!3m2!1d44.439018999999995!2d26.0515156!4m3!3m2!1d44.4397981!2d26.051081699999997!4m3!3m2!1d44.4400312!2d26.0518355!4m3!3m2!1d44.4399332!2d26.0523311!4m3!3m2!1d44.4396554!2d26.052215!5e0!3m2!1sro!2sro!4v1679949014977!5m2!1sro!2sro");
+  const [allForOneText1, setAllForOneText1] = useState("Cursa All For One este o tură populară, necompetitivă, perfectă pentru cei care vor să facă mișcare în natură pe o distanță accesibilă. Aceasta se adresează tuturor persoanelor, de orice vârstă sau gen, chiar și persoanelor cu dizabilități.");
+  const [allForOneText2, setAllForOneText2] = useState("Tocmai în spiritul caritabil din care s-a și născut Aleargă Pentru Viață, participanții sunt încurajați să ofere drept donație pentru Florin cât consideră.");
```
Comment (AMihneaa) [inline client/src/components/CurseHome.tsx:13]: Having too many useState calls suggests you should aim for fewer. You can wrap them in a single state and think smarter about your update commands. Your code lacks modularity; hence, the component should have been designed differently from the start

File: client/src/components/CurseHome.tsx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+        break;
+      case 'Copii':
+        setStartTime('Ora de start: 11:10');
+        setDistance('DISTANȚĂ: 400M');
+        setMinimumDonation('Donatie minima: 20 RON');
+        setAllForOneText1("");
+        setAllForOneText2("");
+        setMapSrc("https://www.google.com/maps/embed?pb=!1m42!1m12!1m3!1d1051.7564164316468!2d26.050924936618955!3d44.43943916368305!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!4m27!3e2!4m5!1s0x40b201c187a44aeb%3A0x776c5045f461c6b0!2sBucure%C5%9Fti%20Sectorul%206%2C%20Bucure%C8%99ti%20061344!3m2!1d44.4396542!2d26.052221199999998!4m3!3m2!1d44.439018999999995!2d26.0515156!4m3!3m2!1d44.4397981!2d26.051081699999997!4m3!3m2!1d44.4400312!2d26.0518355!4m3!3m2!1d44.4399332!2d26.0523311!4m3!3m2!1d44.4396554!2d26.052215!5e0!3m2!1sro!2sro!4v1679949014977!5m2!1sro!2sro");
+        break;
```
Comment (AMihneaa) [inline client/src/components/CurseHome.tsx:34]: The component lacks modularity.
I'd like it to be able to receive an external parameter of type JSON with a unique type defined within it.

File: client/src/components/CurseHome.tsx [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+                        </button>
+                        <button
+                        className="bg-[#006470] hover:bg-teal-500  rounded-md md:text-right sm:text-center lg:px-35 lg:py-33 md:px-20 md:py-20 px-20 py-8"
+                        onClick={() => handleCategoryChange('Copii')}
+                        >
+                        <span className="font-bold lg:text-24 md:text-20 sm:text-20 text-white  whitespace-nowrap">COPII</span>
+                            <br/>
+                        <span className="lg:text-16 md:text-14 text-white whitespace-nowrap">&lt; 10 ANI</span>
+                        </button>
+                        <button
```
Comment (AMihneaa) [inline client/src/components/CurseHome.tsx:102]: All of these have the same structure, so there should be just one where you map based on what you have in the JSON. It's unnecessary to change the characteristics of the component like that.

File: client/src/components/CurseHome.tsx [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+                        >
+                        <span className="font-bold lg:text-24 md:text-20 sm:text-20 text-white  whitespace-nowrap">MASCULIN</span>
+                            <br/>
+                        <span className="lg:text-16 md:text-14 text-white whitespace-nowrap">35+ ANI</span>
+                        </button>
+                    </div>
+                </div>
+                <div className="md:w-3/4 border-4 border-white pr-8 pb-15">
+                    <div className="text-left">
+                        {/* <h2 className="text-3xl font-bold">{selectedCategory}</h2> */}
```
Comment (AMihneaa) [inline client/src/components/CurseHome.tsx:154]: Remove redundant comments from the code; they make it harder to read.

File: client/src/views/Home.view.tsx [hunk 9]
```diff
@@ -11,16 +12,16 @@ const Home = () => {
...
     <>
       <Header />
-      <Banner
+      {/* <Banner
         firstTitle={"Aleargă pentru viață"}
         secondTitle={"APV"}
         video={VideoBanner}
         LseBanner={LseBanner}
         maiputernici={false}
-      />
```
Comment (AMihneaa) [inline client/src/views/Home.view.tsx:20]: Remove redundant comments from the code; they make it harder to read.

File: client/tailwind.config.js [hunk 10]
```diff
@@ -10,6 +10,8 @@ export default {
         'custom-blue': '#1D1A3F',
         'custom-gray': '#656372',
         'custom-green': '#006470',
+        'customBackground': 'rgb(247, 253, 253)',
+
```
Comment (AMihneaa) [inline client/tailwind.config.js:14]: "Go for a darker shade.

File: client/tailwind.config.js [hunk 11]
```diff
@@ -27,6 +29,24 @@ export default {
...
+        '14': '14px',
+        '16': '16px',
+        '20': '20px',
+        '24': '24px',
+        '52': '52px',
+        '40': '40px',
+        '36': '36px',
+        '28': '28px',
+      },
+
```
Comment (AMihneaa) [inline client/tailwind.config.js:49]: Don't use 'px', but 'rem' instead.

File: client/src/components/CurseHome.tsx [hunk 12]
```diff
@@ -0,0 +1,190 @@
+import React, { useState } from 'react';
+import RegulamentAPV2023 from "../assets/RegulamentAPV2023.pdf";
+import pin from "../assets/pin.webp";
+import flyingImg4 from "../assets/flyingImg4.png";
```
Comment (AMihneaa) [inline client/src/components/CurseHome.tsx:4]: Find shorter, more accurate and intuitive names

**Review by AMihneaa — CHANGES_REQUESTED** [review]
The component seems to be generated with ChatGPT. It's good to use it, but it's preferable not to rely solely on it. The styling of the component is lacking; it lacks the human touch that can perceive and understand.

**Pull request closed without merging**